pkg/splunk/enterprise/indexercluster.go

```diff
@@ -718,9 +915,28 @@ func getIndexerClusterList(ctx context.Context, c splcommon.ControllerClient, cr
 }
 
 //RetrieveCMSpec finds monitoringConsole ref from cm spec
-func RetrieveCMSpec(ctx context.Context, client splcommon.ControllerClient, cr *enterpriseApi.IndexerCluster, clusterMasterRef string) (string, error) {
+func RetrieveCMSpec(ctx context.Context, client splcommon.ControllerClient, cr *enterpriseApi.IndexerCluster) (string, error) {
+
+	if len(cr.Spec.ClusterManagerRef.Name) == 0 && len(cr.Spec.ClusterMasterRef.Name) > 0 {
```

Comment:
Can be clubbed into one function. Something like
```
if len(cr.Spec.ClusterManagerRef.Name) == 0 && len(cr.Spec.ClusterMasterRef.Name) > 0 {
		namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: cr.Spec.ClusterManagerRef.Name}
	} else {
		namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: cr.Spec.ClusterMasterRef.Name}
	}
```

Reply:
Good point. The blocker here was `cmCR` type that needs to be specific for `client.Get()`:

**Manager CRD:**
```
	var cmCR enterpriseApi.ClusterManager
	err := client.Get(ctx, namespacedName, &cmCR)
```
**Master CRD:**
```
	var cmCR enterpriseApi.ClusterManager
	var monitoringConsoleRef string = ""
```

That said, I think there's some room here for optimization. Let me update in the next commit.

Reply:
Updates the changes here: https://github.com/splunk/splunk-operator/pull/795/commits/cb111878c0818e1a9ca1bdfc97ae287f53df61f8

We can't fully optimize because `cmCR.Spec.MonitoringConsoleRef.Name` also requires to be type specific.